src/test/java/com/easypost/InsuranceTest.java

```diff
@@ -0,0 +1,87 @@
+package com.easypost;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.model.Insurance;
+import com.easypost.model.InsuranceCollection;
+import com.easypost.model.Shipment;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class InsuranceTest {
+    private static Insurance insurance;
+    private static Map<String, Object> params = new HashMap<>();
+
+    /** 
+     * Setup the testing environment for this file.
+     *
+     * @throws EasyPostException when the request fails.
+     */    
+    @BeforeAll
+    public static void setup() throws EasyPostException {
+        EasyPost.apiKey = System.getenv("EASYPOST_TEST_API_KEY");
+
+        Shipment shipment = Shipment.create(Fixture.oneCallBuyShipment());
+
+        params.put("to_address", Fixture.basicAddress());
+        params.put("from_address", Fixture.basicAddress());
+        params.put("tracking_code", shipment.getTrackingCode());
+        params.put("carrier", Fixture.usps());
+        params.put("amount", "100");
+
+        insurance = Insurance.create(params);
+    }
+
+    /**
+     * Test creating an insurance object.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreate() throws EasyPostException {
+        Insurance insurance = Insurance.create(params);
+
+        assertTrue(insurance instanceof Insurance);
```

Comment:
Please add an assertion to check the amount field and make sure it matches (sorry I didn't do this in the PHP lib)